third_party/terraform/resources/resource_logging_bucket_config.go

```diff
@@ -0,0 +1,183 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+var loggingBucketConfigSchema = map[string]*schema.Schema{
+	"name": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"location": {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+	"bucket_id": {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+	"description": {
+		Type:     schema.TypeString,
+		Optional: true,
+		Computed: true,
+	},
+	"retention_days": {
+		Type:     schema.TypeInt,
+		Optional: true,
+		Default:  30,
+	},
+	"lifecycle_state": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+}
+
+type loggingBucketConfigIDFunc func(d *schema.ResourceData, config *Config) (string, error)
+
+// ResourceLoggingBucketConfig creates a resource definition by merging a unique field (eg: folder) to a generic logging bucket
+// config resource. In practice the only difference between these resources is the url location.
+func ResourceLoggingBucketConfig(parentType string, parentSpecificSchema map[string]*schema.Schema, iDFunc loggingBucketConfigIDFunc) *schema.Resource {
+	// func ResourceLoggingBucketConfig() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceLoggingBucketConfigAcquire(iDFunc),
+		Read:   resourceLoggingBucketConfigRead(),
+		Update: resourceLoggingBucketConfigUpdate(),
+		Delete: resourceLoggingBucketConfigDelete(),
+		Importer: &schema.ResourceImporter{
+			State: resourceLoggingBucketConfigImportState(parentType),
+		},
+		Schema: mergeSchemas(loggingBucketConfigSchema, parentSpecificSchema),
+	}
+}
+
+var loggingBucketConfigIDRegex = regexp.MustCompile("(.+)/(.+)/locations/(.+)/buckets/(.+)")
+
+func resourceLoggingBucketConfigImportState(parent string) schema.StateFunc {
+	return func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+		// loggingSinkId, err := parseLoggingSinkId()
+		parts := loggingBucketConfigIDRegex.FindStringSubmatch(d.Id())
+		if parts == nil {
+			return nil, fmt.Errorf("unable to parse logging sink id %#v", d.Id())
+		}
+
+		if len(parts) != 5 {
+			return nil, fmt.Errorf("Invalid id format. Format should be '{{parent_type}}/{{parent_id}}/locations/{{location}}/buckets/{{bucket_id}} with parent_type in %s", loggingSinkResourceTypes)
```

Comment:
You added the parent type as a parameter to this func, so you could just use it  here and below instead of checking the array of possible types

Reply:
Parent is the terraform formate of the variable, eg: `billing_account`, so it wouldn't make sense in the error formatting. I could add both `billing_account` and `billingAccount` formats but for just error formatting i thought it wasn't worth it. FWIW the docs have per-type import formats.